nshutijonathan/Auto-mart3
# #166546064 Build out seller post car advert endpoint in  Auto-Mart project

#### What does this PR do?
Have created Car advert endpoint
Have the Cars advert endpoints validated
#### Description of Task to be completed?
 - Validated the inputs
- Implemented Authorizations
- Protected  create cars advert Route
#### Have The following endpoint working and validated
*   POST/localhost:3000/api/v2/auth/signup
*   POST/localhost:3000/api/v2/auth/signin
*   POST/localhost:3000/api/v2/auth/signup/admin
*   GET/localhost:3000/api/v2/users
*   GET/localhost:3000/api/v2/users/<:id>
*   DELETE/localhost:3000/api/v2/users/<:id>
*   POST/localhost:3000/api/v2/car
*   GET/localhost:3000/api/v2/cars
#### How should this be manually tested?
* Download or clone the tutorial project code from [this repo](https://github.com/nshutijonathan/Auto-Mart3), Install all required NPM  packages by running NPM install from the command line in the project root folder (where the package.json is located).
* Start the API by running NPM  start from the command line in the project root folder, you should see the message Server listening on port 3000. You can test the API directly using an application such as Postman or you can test it with one of the single pages in the browser.
* You must have SQL database installed and .env file in the root folder
* to test authentication and authorization, add this to the header 
Key: :x-auth-token,value:JWT TOKEN
#### Any background context you want to provide?
* There is Implementation of Javascript(ES6) logic with SQL database  to persist some data
#### What are the relevant pivotal tracker stories?
[#166546064](https://www.pivotaltracker.com/n/projects/2346053/stories/166546064)

#### Screenshots
![car_advert3](https://user-images.githubusercontent.com/16412410/59148705-9543cd00-8a0c-11e9-81fe-85e5f9973e3a.PNG)

####Questions
--
